tree.go

```diff
@@ -349,7 +349,12 @@ func (n *node) insertChild(path string, fullPath string, handlers HandlersChain)
 		}
 
 		if len(n.path) > 0 && n.path[len(n.path)-1] == '/' {
-			panic("catch-all conflicts with existing handle for the path segment root in path '" + fullPath + "'")
+			pathSeg := strings.SplitN(n.children[0].path, "/", 2)[0]
```

Comment:
I agree, but I prefer the original one to keep the  style consistent, as it is used in other place like [tree.go#L240](https://github.com/gin-gonic/gin/blob/master/tree.go#L240):

```go
// Wildcard conflict
pathSeg := path
if n.nType != catchAll {
	pathSeg = strings.SplitN(pathSeg, "/", 2)[0]
}
```

I've merged from upstream and maybe we can run the test again to see if more test cases should be added, since the `tree_test.go` doesn't check the original panicMsg here before, so the change of message format should not affect the coverage.

Reply:
@SericaLaw Now the Codecov has become Green, I think the coverage removal before has to do with Codecov itself. #2952 merged the coverage removal so your PR can pass Codecov.

Reply:
@Bisstocuz Thanks, what's your further opinion about the commit? If you agree with my thought, please resolve the conversation.

Reply:
@SericaLaw No further opinion, and "resolve" button is on your side, you can click it now.